lib/chef/server.rb

```diff
@@ -18,7 +18,14 @@ def self.from_chef_server_url(url)
     def parse_server_version(line)
       # first line from the /version endpoint will either be this format "chef-server 12.17.5\n"
       # or, when habitat, this format "Package: chef-server/chef-server-nginx/12.17.42/20180413212943\n"
-      Gem::Version.new(line.include?('/') ? line.split('/')[2] : line.split(' ').last.gsub(/\+.*$/, ''))
+      version_str = if line.include?('/')
+                      line.split('/')[2]
+                    else
+                      # Use partition instead of regex to avoid ReDoS vulnerability
```

Comment:
ReDos is not a thing from Ruby 3.2 onwards so there's no real need to do this in modern Ruby